### PR TITLE
RNA-Seq Workflow v2.0

### PR DESCRIPTION
**Pull request by claymcleod**
[Rendered](https://github.com/stjude/sjcloud-rfcs/blob/cmcleod/rnaseq-workflow-v2/text/0000-rnaseq-workflow-v2.0.md)

### CODE REVIEW

File: text/0000-rnaseq-workflow-v2.0.md [hunk 1]
```diff
@@ -0,0 +1,478 @@
...
+>   * Last, the various other pipelines in the community don't tend to remove these features:
+>     * The [GDC documentation](https://docs.gdc.cancer.gov/Data/Bioinformatics_Pipelines/Expression_mRNA_Pipeline/#rna-seq-alignment-command-line-parameters) does not currently describe any filtering of their [GENCODE v22][gencode-v22] GTF (see the section labeled "Step 1: Building the STAR index").
+>     * ENCODE is not filtering any features, they are just changing some of the contig names in the [GENCODE v24][gencode-v24] GTF (see the analysis [in the appendix](#ENCODE-GTF-generation)).
+>     * The TOPMed pipeline [does outline some postprocessing](https://github.com/broadinstitute/gtex-pipeline/blob/master/TOPMed_RNAseq_pipeline.md#reference-annotation) they are doing to the [GENCODE v26][gencode-v26] GTF, but it's mostly just collapsing exons. I inspected the script, which does have some functionality built in to blacklist a list of transcript IDs. However, the documentation does not specify that this is turned on by default.
+
+After discussion internally, we decided to continue removing `level 3` annotations by default:
```
[the remark this answers is not in the document]
Comment (claymcleod) [reply text/0000-rnaseq-workflow-v2.0.md]: Yeah, I agree. I think this is the cleanest solution. I say we move forward with this approach.

**Comment by claymcleod** [comment]
> > @adthrasher For some reason, I can't reply directly on your comment above...
> > What do you think about sha256 vs BLAKE2b (see Michael M's comment above)? If the coreutils suggests the latter, maybe we should use it instead... I don't know. I do think three would be overboard.
> 
> It depends.
> 
> For the end user, I think they're only likely to use md5 (if anything). Only a small set of users is likely to use sha256.
> 
> For internal usage, going with a more cryptographically secure method is preferable, but I don't think we should generate three checksums. What internal use cases do we have for the checksums? That may give some clarity on the best choice.

At this point, I think generating MD5s only would be sufficient, but we may also want to generate SHA256 internally because I think that is a better long-term solution.

**Comment by adthrasher** [comment]
> @adthrasher, thoughts? ^

I agree with Alex that this is a good idea to include. I'll take a look at what is involved.

**Comment by adthrasher** [comment]
The two week public comment period has elapsed. So this is considered final. Any further changes can be made through a new PR.

**Comment by adthrasher** [comment]
Assigned RFC #0001

**Comment by kmhernan** [comment]
Just saw this and wanted to say the GDC absolutely does not mark duplicates for RNA-Seq. That is generally not a good practice for RNA-Seq unless you have UMI where you can confidently remove duplicates. We also run all the readgroups at once through star (you have to do SE and PE separately though).

**Comment by claymcleod** [comment]
Hi @kmhernan, thanks for your comment on this. Note that we are only marking the duplicates (not removing them). I'm curious if you are aware of any negative downstream effects from just marking duplicates for RNA-Seq data? Most tools that should account for these duplications (e.g. expression quantification using `htseq`) ignore the flag (at least to my knowledge).

If true, that would make the decision whether to mark the duplicates or not one of opinion. After giving it a few moments thought, it may actually be better to assert nothing at all about the duplicate status. That would also reduce the runtime, which is a plus.

For those interested, this is worth skimming: https://www.nature.com/articles/srep25533.

> Currently, quantitative RNA-seq methods are pushed to work with increasingly small starting amounts of RNA that require amplification. However, it is unclear how much noise or bias amplification introduces and how this affects precision and accuracy of RNA quantification.... We find that a large fraction of computationally identified read duplicates are not PCR duplicates and can be explained by sampling and fragmentation bias. Consequently, the computational removal of duplicates does improve neither accuracy nor precision and can actually worsen the power and the False Discovery Rate (FDR) for differential gene expression. Even when duplicates are experimentally identified by unique molecular identifiers (UMIs), power and FDR are only mildly improved...

I'll discuss with our team. Thanks again for your contribution!

cc: @adthrasher

**Comment by adthrasher** [comment]
After internal consideration, we have decided to no longer mark duplicates for RNA-Seq. The pipeline will be updated in the near future.

**Comment by kmhernan** [comment]
Hey sorry was traveling this weekend. Yeah just marking seems like it wouldn’t be an issue (only compute cost). HTSeq does some bizarro things that you can only really see in source (at least from my experience). I actually prefer just getting the counts from STAR cause kallisto Salmon etc seem to perform better in quantification anyways. If you wanted to call rna-seq variants (no good way to do this yet) then your markdups step would be needed.

**Comment by hbeale** [comment]
Hi folks - Thanks for inviting feedback. As you know, we at Treehouse have many similar issues to those you address here (1). We agree that updating reference files and software is worthwhile. We do not perform gene model post-processing for the reasons you've outlined. We include potential duplicate reads in our quantification steps, but subsequently mark duplicates and count non-duplicate reads as a QC step. We have found that counting mapped, exonic, non-duplicate reads has allowed us to identify and exclude some extremely low quality samples that would not otherwise be detected (2,3)

1 Vaske et al, 2019, http://dx.doi.org/10.1001/jamanetworkopen.2019.13968
2 https://github.com/UCSC-Treehouse/mend_qc
3 https://www.biorxiv.org/content/10.1101/716829v1

**Comment by claymcleod** [comment]
Thanks for reviewing @hbeale, this is really helpful and affirming feedback. We'll be sure to let you know when we release the new version of the data.

**Pull request merged**